frontend/src/js/ui-components/InputDate.tsx

```diff
@@ -53,13 +59,22 @@ const InputDate = forwardRef<HTMLInputElement, Props>(
       onChange,
       onCalendarSelect,
       onFocus,
+      onBlur,
+      onClick,
       ...props
     },
     ref,
   ) => {
     const datePickerRef = createRef<ReactDatePicker>();
+    const [hasFocus, setHasFocus] = useState(false);
     const [focusBlocked, setFocusBlocked] = useState(false);
 
+    useEffect(() => {
+      if (props.inputProps?.autoFocus && hasFocus) {
```

Comment:
Hmm, I think while `onClick` of the input should open the picker, we shouldn't auto-open on `autoFocus`, because that feels like "a modal the user didn't trigger". So can we maybe live without the useEffect?